### ISSUE TITLE
Swagger-Editor won´t start locally

### ISSUE THREAD

**Issue by MrMojoR** [swagger-api/swagger-editor#535]
I have installed node.js, I´ve followed the instructions, but I couldn´t get it to work. Something is wrong with grunt, I think. Here are the output, and the log file:

Output:
```
Running "connect:dist:keepalive" (connect) task
Waiting forever...
Started connect web server on http://localhost:8080
Fatal error: spawn cmd ENOENT


Execution Time (2015-07-08 12:47:08 UTC)
less:server             782ms  █ 2%
concurrent:dist          8.7s  █████████ 23%
ngAnnotate:dist          2.9s  ███ 8%
cssmin:generated        433ms  █ 1%
uglify:generated          24s  ████████████████████████ 63%
connect:dist:keepalive  463ms  █ 1%
Total 38.1s


npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs
\\node_modules\\npm\\bin\\npm-cli.js" "start"
npm ERR! node v0.12.6
npm ERR! npm  v2.11.2
npm ERR! code ELIFECYCLE
npm ERR! swagger-editor-src@2.9.6 start: `grunt serve:dist`
npm ERR! Exit status 3
npm ERR!
npm ERR! Failed at the swagger-editor-src@2.9.6 start script 'grunt serve:dist'.

npm ERR! This is most likely a problem with the swagger-editor-src package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     grunt serve:dist
npm ERR! You can get their info via:
npm ERR!     npm owner ls swagger-editor-src
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\EMRTOSV\swagger-editor\npm-debug.log
C:\Users\EMRTOSV\swagger-editor [master]>

npm-debug.log:
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start' ]
2 info using npm@2.11.2
3 info using node@v0.12.6
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info prestart swagger-editor-src@2.9.6
6 verbose unsafe-perm in lifecycle true
7 info start swagger-editor-src@2.9.6
8 verbose unsafe-perm in lifecycle true
9 info swagger-editor-src@2.9.6 Failed to exec start script
10 verbose stack Error: swagger-editor-src@2.9.6 start: `grunt serve:dist`
10 verbose stack Exit status 3
10 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:213:16)
10 verbose stack     at EventEmitter.emit (events.js:110:17)
10 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
10 verbose stack     at ChildProcess.emit (events.js:110:17)
10 verbose stack     at maybeClose (child_process.js:1015:16)
10 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
11 verbose pkgid swagger-editor-src@2.9.6
12 verbose cwd C:\Users\EMRTOSV\swagger-editor
13 error Windows_NT 6.1.7601
14 error argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
15 error node v0.12.6
16 error npm  v2.11.2
17 error code ELIFECYCLE
18 error swagger-editor-src@2.9.6 start: `grunt serve:dist`
18 error Exit status 3
19 error Failed at the swagger-editor-src@2.9.6 start script 'grunt serve:dist'.
19 error This is most likely a problem with the swagger-editor-src package,
19 error not with npm itself.
19 error Tell the author that this fails on your system:
19 error     grunt serve:dist
19 error You can get their info via:
19 error     npm owner ls swagger-editor-src
19 error There is likely additional logging output above.
20 verbose exit [ 1, true ]
```
The ports are surely free. Thank you very muck for your help!

**Comment by mohsen1** [comment]
Just to make sure do a deep clean up and start again

```
rm -rf node_modules app/bower_components
bower cache clean
npm cache clean
npm start
```

**Comment by MrMojoR** [comment]
Unfortunately this hasn't solved the problem. Could any other software interfere with this application?
Thank you for your answer!

**Comment by mohsen1** [comment]
Editor will try to run on port `8080`. If that port is busy try changing it in [connect.js](https://github.com/swagger-api/swagger-editor/blob/master/grunt/connect.js#L4) or set the environment variable `PORT` to something else

**Comment by MrMojoR** [comment]
Sorry for the late reply. It‘s still not working. Do you have other ideas? Thank you!

**Comment by mohsen1** [comment]
"It's not working" is not helpful. What is not working? What happens when you `npm start`?

**Comment by MrMojoR** [comment]
When I type in npm start, I get the same error message over and over again, which I have already attached.

**Comment by mohsen1** [comment]
I have no idea why `ELIFECYCLE` is being thrown. Can you try installing `grunt` and `bower` globally using npm and run `grunt serve`?

**Comment by MrMojoR** [comment]
I have already tried that, the result was also the same.

**Comment by mohsen1** [comment]
did you delete the folder, cloned again and tried starting again?

I see `Fatal error: spawn cmd ENOENT` in your logs. Apparently someone can't make a directory. It could be permission issues on your system.

**Comment by MrMojoR** [comment]
Of course, I've tried that. I have also tried starting cmd with administrator privileges.

**Comment by mohsen1** [comment]
Honestly at this point I don't know how to help. I have a Windows VM. I'll try running it in my VM to see what happens

**Comment by MrMojoR** [comment]
I have also tried it in a Windows 7 (32 bit) VM, it worked perfectly. I think the problem is with this particular PC, which by the way also running Windows 7 64 bit. jBoss application server might be interfering, what do you think?

**Comment by chrisdudley** [comment]
Try stopping jBoss? I certainly found that having eclipse running caused issues as it was using the default port.
For what its worth I just did a clone of git master and npm start worked fine on win 7 64bit.

**Comment by MrMojoR** [comment]
I have stopped jBoss, but it's still the same. Thank you very much guys for your efforts!

**Comment by mohsen1** [comment]
I successfully ran Swagger Editor build in a Windows VM.

Things that I notices:

* I had JDK installed in my Windows VM. Selenium dependency of Swagger Editor needs JDK but in the build of `npm start` you don't need it. Installing JDK can help.
* I ran `npm start` in regular Windows CMD application, not Git Bash.
* I had `git`, `node` and `npm` in my path
* npm `1.4.3`
* node `0.10.26`

Your Windows machines can be different, npm and node versions of my Windows machine are pretty old but that means it even works with older version. Unless they broke something in newer version of npm or node in Windows!
